src/View/mainpage/DrawROIWindow/Drawing.py

```diff
@@ -96,14 +103,31 @@ def _display_pixel_color(self):
             creates a new layer over the existing image instead of replacing 
             it. """
             # Convert QPixMap into Qimage
-            for x_coord, y_coord in self.target_pixel_coords:
-                c = self.q_image.pixel(x_coord, y_coord)
-                colors = QColor(c).getRgbF()
-                self.according_color_dict[(x_coord, y_coord)] = colors
+            if self.different_sizes:
+                for x_coord, y_coord in self.target_pixel_coords:
```

Comment:
I think it would be even better if you can move this if else statement outside of the for loop because the dataset['Rows'].value is the same for all point in contour. So it is a bit redundant and time-consuming if the program have to check this for every point in the contour.

_Originally posted by @DuyTungHa in https://github.com/didymo/OnkoDICOM/pull/160#discussion_r702541600_

I originally had the if statement in the code like that, but as @DuyTungHa pointed out, as the condition doesnt change for each point, keeping the condition outside speeds up the process

Reply:
I see that now... However the trade off for that is this code repetition which is not ideal, especially when the repetition happen across multiple places in the code like this. It makes this file a lot more difficult to maintain... Can you think of anyway to resolve this?

Maybe when opening the drawing window, we transform the x and y coordinates first so that we dont have to re-transform them everytime there's a change in the drawing?

Then when we save the final coordinates, we can transform them back to the correct coordinates for saving?

Reply:
Because if you transform the coordinates in this function, it will transform all drawn coordinates again and again everytime there's a change in the drawing, thus probably will slow down the drawing process for PET scan

Reply:
Currently most of the code hadles and saves the data as n\*n, while 512\*512 is used just to interact with the QImage. Although inv_linear_transform is O(n\*n), linear_transform is O(1). Most of the code to read and check conditions uses linear_transform and is O(1) and inv_linear_transform is only used to write to QImage. I will implement the transform_coord_to_point as that will reduce duplicate code, but thats the best optomisation I can do

Reply:
Thanks!
Yes, tbh I'm more concerned about the duplication since to actually increase the performance we need to refactor it, and it's not in the scope of this PR. Hence I just want the code to be more maintainable so that in the future, it will be less of a  burden for the developer who refactors this function :)